### DISCUSSION TITLE
May I ask how Animate 2023 uses 64 bit programs during debugging?

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#2769] [Q&A]
This is really a headache. For Windows desktops, 64 bit cannot be used during debugging, only 32 bit can be used, which makes some memory intensive programs completely unable to run during debugging until they get stuck.
Is there any way to call a 64 bit system when ctrl+enter?
Thank you!

**Comment by M4rkY4** [comment]
perhaps, you can find 32-bit flash player debugger exe that Animate uses (somewhere in AppData folder or where it is) and rewrite it with a 64-bit one?

**Comment by Jason0247** [comment]
![image](https://github.com/airsdk/Adobe-Runtime-Support/assets/91530077/89b9312b-968c-4c59-b93a-5698d11ae275)

I tried renaming these two files in the sdk directory to use a 64 bit startup program, but after ctrl+enter, the prompt failed. @M4rkY4

**Comment by ajwfrost** [comment]
I think most of the IDEs have the concept of "begin remote debugging session" where the IDE goes into a "waiting for player" status..? If so, you can kick that off, and then go and manually run your application using adl64.exe, and it should then connect.

**Reply by balaedit** [reply]
I have one question regarding this that to test the apk on client device both on android or desktop, creating apk debugging mode and send it to client and then it will open the ip box on client screen. What we have to enter in that box to debug in my pc via remote.

**Reply by ajwfrost** [reply]
If you're getting the pop-up on a device that's asking for the debugger details, then the IP address you enter in there should be the IP address of a computer where the Flash Debugger is sat, in the "waiting for player" mode (i.e. start remote debug session from Animate/IntelliJ/etc). The device and the computer need to both be on the same wifi so that they can connect. If you get this pop-up on a desktop computer, you can type "127.0.0.1" which is the localhost i.e. the same computer (assuming you're debugging on the same machine as the application)

**Comment by Jason0247** [comment]
Do you have any relevant information or steps? I have tried it for a few days but it still doesn't work. When debugging, it is still a 32-bit program.

**Reply by ajwfrost** [reply]
So I just created a file in Animate that just put into a textfield the result of `Capabilities.cpuAddressSize`. Ctrl-Enter gives me a value of 32.

Then I did Debug -> Begin Remote Debug Session
The output pane was `Waiting for player to connect...`
Then I go into a command prompt and run:
```
path\to\air\sdk\bin\adl64.exe info-app.xml
```
Output pane then has: `[SWF] Info.swf - 1051 bytes after decompression` and the display shows 64, so it's running in 64-bit mode..

Note, the publish settings for the SWF must have "Permit debugging" checked for this to work!

thanks

**Comment by Jason0247** [comment]
Thank you first. This information is very useful, and I have found remote debugging.
But I don't know what the command prompt is, it's executed through cmd and entering the command. I'm really sorry to ask such a low-level question. @ajwfrost

**Reply by balaedit** [reply]
Sorry for not sharing details as currently pc is not available. Will share some Adobe docs. I will share in details as soon I reach.

https://help.adobe.com/en_US/air/build/WSfffb011ac560372f-6fa6d7e0128cca93d31-8000.html

**Reply by ajwfrost** [reply]
.. and if your question was even more low-level -> try hitting Windows-R and then entering "cmd", which will get you the command prompt up. You can then use "cd folder_name" to move to where your application is, for example `cd "c:\users\username\documents\my app"` (you can generally copy the path location from the Windows Explorer address bar). Then if you run `dir` you should see the files/folders that are present.

Running the `adl64.exe` binary from here means you need to put the path to this before it (or you can add this path - i.e. the location of the AIR SDK with 'bin' after it - to your Windows environment variables, but you'd need to re-start the command prompt after making a change there). So e.g. "c:\users\username\downloads\airsdk\bin" could be your path, so then you'd have to launch using `c:\users\username\downloads\airsdk\bin\adl64.exe appname-app.xml` plus whatever optional parameters you wanted (see reference in the earlier post above). appname-app.xml would need to be changed to whatever Animate has generated for you there..

Hope that helps :-)

thanks
